File: SYP.Web/Modules/ServerTypes/Catalog/ProductPackingEditorColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ProductPackingRow } from "./ProductPackingRow";

export interface ProductPackingEditorColumns {
    Id: Column<ProductPackingRow>;
    Code: Column<ProductPackingRow>;
    Name: Column<ProductPackingRow>;
    Quantity: Column<ProductPackingRow>;
    IsActive: Column<ProductPackingRow>;
}

export class ProductPackingEditorColumns extends ColumnsBase<ProductPackingRow> {
    static readonly columnsKey = 'Catalog.ProductPackingEditor';
    static readonly Fields = fieldsProxy<ProductPackingEditorColumns>();
}

File: SYP.Web/Modules/Catalog/ProductPacking/Editor/ProductPackingEditorColumns.cs
namespace SYP.Catalog.Columns;

[ColumnsScript("Catalog.ProductPackingEditor")]
[BasedOnRow(typeof(ProductPackingRow), CheckNames = true)]
public class ProductPackingEditorColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Code { get; set; }
    [EditLink]
    public string Name { get; set; }
    public int Quantity { get; set; }
    public bool IsActive { get; set; }
}
